changelog/5063.feature.rst

```diff
@@ -0,0 +1 @@
+Switch from ``pkg_resources`` to ``importlib-metadata`` for entrypoint detection for improved performance and import time.
```

Comment:
```suggestion
Switch from ``pkg_resources`` to ``importlib-metadata`` for entrypoint detection for improved performance and import time.

As a result of this, ``importlib-metadata`` is a new dependency.
```

Reply:
Hmmm it can be argued that this is obvious from the initial description, so feel free to discard this. Up to you. 👍

Reply:
going to just leave it then -- thanks for the review!